pkg/testutils/moduletemplate.go

```diff
@@ -48,6 +48,22 @@ func ModuleTemplateExists(ctx context.Context,
 	return nil
 }
 
+func ModuleTemplateExistsByName(ctx context.Context,
```

Comment:
to reviewer: I was unable to re-use the existing `ModuleTemplateExists` function. It seems as if this existing function relies on the ModuleReleaseMeta to find the module template (which I find weird). In my case, I want to verify that ModuleTemplates are not synced anymore when not referenced in a ModuleReleaseMeta. Hence, I was unable to use the existing function.

I think this needs a cleanup but I would propose to do this in a follow-up as this PR is already quite big.

Reply:
Can you provide the list of such findings as a comment in the issue? I am pretty sure we'll forget about it.